### PR TITLE
Simplify Windows implementation of Compilation::performSimpleCommand

### PR DESCRIPTION
**Pull request by hughbe**
See https://msdn.microsoft.com/en-us/library/eyw7eyfw.aspx for the docs for _putenv_s

I also get the Microsoft Code Analysis warning that
> Warning	C6031	Return value ignored: '_putenv'

I think that the code analysis has a point here, but I'm not sure if I've got the right fix here. Now, if we can't set the environment variable, we return the error code instead of continuing with compilation as we did before.

Perhaps there are better options
- assert that the error code is 0, as we expect this to always work
- print some kind of informative message that we failed (a warning?), and continue compilation

@jrose-apple as the code owner of swiftDriver what do you think?

### CODE REVIEW

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Review by jrose-apple — COMMENTED** [review]
I'd *like* to say we should assert because it increases our chances of recovery. Maybe we can assert in debug builds and bail out early in release builds?

Thanks for fixing this, by the way. I guess none of us are Windowsy enough to know about `_putenv_s`.

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Comment by hughbe** [comment]
@swift-ci please smoke test OSX Platform

**Comment by hughbe** [comment]
@swift-ci please smoke test OS X Platform

**Comment by hughbe** [comment]
Looks like it's not impossible for `setenv` to return anything but 0... what do you think we should do?

File: lib/Driver/Compilation.cpp [hunk 1]
```diff
@@ -747,13 +747,16 @@ int Compilation::performSingleCommand(const Job *Cmd) {
...
-    llvm::SmallString<256> envStr = StringRef(envPair.first);
-    envStr.append(StringRef("="));
-    envStr.append(StringRef(envPair.second));
-    _putenv(envStr.c_str());
+    int envResult =_putenv_s(envPair.first, envPair.second);
 #else
-    setenv(envPair.first, envPair.second, /*replacing=*/true);
+    int envResult = setenv(envPair.first, envPair.second, /*replacing=*/true);
 #endif
+    assert(envResult &&
```
Comment (jrose-apple) [inline lib/Driver/Compilation.cpp]: You got your assertion backwards. :-)

Comment (hughbe) [inline lib/Driver/Compilation.cpp]: Ahhhh facepalm

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Pull request merged**